keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -531,20 +532,30 @@ the kubelet, as described below. However, the source of this data may vary; for
 example, a cloud provider controller could populate this based upon information
 from the cloud provider API.
 
-In the kubelet case, each kubelet publishes kubelet publishes a set of
-`ResourceSlice` objects to the API server with content provided by the
-corresponding DRA drivers running on its node. Access control through the node
-authorizer ensures that the kubelet running on one node is not allowed to
-create or modify `ResourceSlices` belonging to another node. A `nodeName`
-field in each `ResourceSlice` object is used to determine which objects are
-managed by which kubelet.
-
-**NOTE:**  `ResourceSlices` are published separately for each driver, using
-whatever version of the `resource.k8s.io` API is supported by the kubelet. That
-same version is then also used in the gRPC interface between the kubelet and
-the DRA drivers providing content for those objects. It might be possible to
-support version skew (= keeping kubelet at an older version than the control
-plane and the DRA drivers) in the future, but currently this is out of scope.
+In the kubelet case, each driver running on a node publishes a set of
+`ResourceSlice` objects to the API server for its own resources, using its
+connection to the apiserver. Access control through a validating admission
+policy can ensure that the drivers running on one node are not allowed to
+create or modify `ResourceSlices` belonging to another node. The `nodeName`
+and `driverName` fields in each `ResourceSlice` object are used to determine which objects are
+managed by which driver instance. The owner reference ensures that objects
+beloging to a node get cleaned up when the node gets removed.
+
+In addition, whenever kubelet starts, it first deletes all `ResourceSlices`
+belonging to the node with a `DeleteCollection` call that uses the node name in
+a field filter. This ensures that no pods depending in DRA get scheduled to the
+node until the required DRA drivers have started up again (node reboot) and
+reconnected to kubelet (kubelet restart). It also ensures that drivers which
+don't get started up again at all don't leave stale `ResourceSlices`
+behind. Garbage collection does not help in this case because the node object
+still exists. For the same reasons, the ResourceSlices belonging to a driver
+get removed when the driver unregisters, this time with a field filter for node
+name and driver name.
+
+Deleting `ResourceSlices` is possible because all information in them can be
```

Comment:
If there was status in the Pool, is it OK to lose that?

Reply:
This is something that we need to update when we add such status. I'd like to iterate on the KEP by merging small updates once they are ready.

I can also prepare one big PR with several commits, but then the discussion becomes less focused.